input/collector_config.go

```diff
@@ -0,0 +1,53 @@
+package input
+
+import (
+	"os"
+
+	"github.com/pganalyze/collector/config"
+	"github.com/pganalyze/collector/state"
+)
+
+func getCollectorConfig(c config.ServerConfig) state.CollectorConfig {
+	return state.CollectorConfig{
+		SectionName:             c.SectionName,
+		DisableLogs:             c.DisableLogs,
+		DisableActivity:         c.DisableActivity,
+		EnableLogExplain:        c.EnableLogExplain,
+		DbName:                  c.DbName,
+		DbUsername:              c.DbUsername,
+		DbHost:                  c.DbHost,
+		DbPort:                  int32(c.DbPort),
+		DbSslmode:               c.DbSslMode,
+		DbHasSslrootcert:        c.DbSslRootCert != "" || c.DbSslRootCertContents != "",
+		DbHasSslcert:            c.DbSslCert != "" || c.DbSslCertContents != "",
+		DbHasSslkey:             c.DbSslKey != "" || c.DbSslKeyContents != "",
+		DbExtraNames:            c.DbExtraNames,
+		DbAllNames:              c.DbAllNames,
+		AwsRegion:               c.AwsRegion,
+		AwsDbInstanceId:         c.AwsDbInstanceID,
+		AwsHasAccessKeyId:       c.AwsAccessKeyID != "",
+		AzureDbServerName:       c.AzureDbServerName,
+		AzureEventhubNamespace:  c.AzureEventhubNamespace,
+		AzureEventhubName:       c.AzureEventhubName,
+		AzureAdTenantId:         c.AzureADTenantID,
+		AzureAdClientId:         c.AzureADClientID,
+		AzureHasAdCertificate:   c.AzureADCertificatePath != "",
+		GcpCloudsqlInstanceId:   c.GcpCloudSQLInstanceID,
+		GcpPubsubSubscription:   c.GcpPubsubSubscription,
+		GcpHasCredentialsFile:   c.GcpCredentialsFile != "",
+		GcpProjectId:            c.GcpProjectID,
+		ApiSystemId:             c.SystemID,
+		ApiSystemType:           c.SystemType,
+		ApiSystemScope:          c.SystemScope,
+		DbLogLocation:           c.LogLocation,
+		DbLogDockerTail:         c.LogDockerTail,
+		IgnoreTablePattern:      c.IgnoreTablePattern,
+		IgnoreSchemaRegexp:      c.IgnoreSchemaRegexp,
+		QueryStatsInterval:      int32(c.QueryStatsInterval),
+		MaxCollectorConnections: int32(c.MaxCollectorConnections),
```

Comment:
Was not sure if I should have the int32 casts here or in `transform`--it looks like `state` is using both `int` and `int32` for different things.

Reply:
Either or works - no strong preference one way or another. We could probably adjust the type for these directly in the config settings as well, so we use `int32` throughout. Though I'll note that technically these could be unsigned int32.

Reply:
Okay, I'll leave it for now and we can adjust types later.